File: code/c/Code_5_09_1.c
#include <stdio.h>

int main() {
	// 輸入
	long long N;
	scanf("%lld", &N);

	// 支付方式的模擬
	long long answer = 0;
	while (N >= 10000) {
		N -= 10000;
		answer += 1;
	}
	while (N >= 5000) {
		N -= 5000;
		answer += 1;
	}
	while (N >= 1) {
		N -= 1000;
		answer += 1;
	}
	
	// 輸出答案
	printf("%lld\n", answer);
	return 0;
}

File: code/python/Code_5_09_1.py
N = int(input())

answer = 0

# 10000 圓以上則繼續使用 10000 圓紙鈔
while N >= 10000:
	N -= 10000
	answer += 1

# 5000 圓以上則繼續使用 5000 圓紙鈔
while N >= 5000:
	N -= 5000
	answer += 1

# 剩餘金額使用 1000 圓紙鈔支付
while N >= 1:
	N -= 1000
	answer += 1

print(answer)
